src/master-playlist-controller.js

```diff
@@ -711,24 +711,38 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return false;
     }
 
-    // when the video is a live stream
-    if (!media.endList) {
+    // when the video is a live stream, or has a specific start time:
+    if (!media.endList || media.start) {
       const seekable = this.seekable();
 
       if (!seekable.length) {
-        // without a seekable range, the player cannot seek to begin buffering at the live
-        // point
+        // without a seekable range, the player cannot seek to a specific time
         return false;
       }
 
+      const seekableTime = seekable.end(0);
+      let startTime = seekableTime;
+
+      if (media.start) {
+        const offset = media.start.timeOffset;
+
+        if (offset < 0) {
+          startTime += offset;
+        } else {
+          startTime = offset;
+        }
+        // cap the startTime to within the seekable range:
+        startTime = Math.max(seekable.start(0), Math.min(seekableTime, startTime));
```

Comment:
By https://tools.ietf.org/html/draft-pantos-http-live-streaming-23#section-4.3.5.2 , I think we need to account for `PRECISE`, or at least mention it in a comment (it's probably OK to ignore using it and just add in a comment for the time being, since this PR just makes things better)